features/FEATURE_BLE/source/Gap.cpp

```diff
@@ -296,8 +302,14 @@ void Gap::processConnectionEvent(
     const uint8_t *localResolvableAddr
 ) {
     /* Update Gap state */
-    state.advertising = 0;
-    state.connected   = 1;
+    if (role == PERIPHERAL) {
+        state.advertising = 0;
+    } else {
+        MBED_ASSERT(role == CENTRAL);
+        stopRadioScan();
```

Comment:
Normally, scan stops when connection is initiated.

Reply:
If so, then we may need to re-implement the “connect“ method.

Reply:
It's break-out change.

Reply:
I should have been more explicit but yes that's the route we should take I think. The same pattern applied to stopScan with can be applied to connect.

Reply:
Although this is somewhat difficult, it seems to be the best solution. I try to implement it.